lib/cocoapods-core/source/aggregate.rb

```diff
@@ -198,26 +187,34 @@ def update_search_index(search_data)
       # @param  [Set] set
       #         The set for which the information is needed.
       #
-      # @note   If the specification can't load an empty hash is returned and
+      # @note   If the specification can't load an empty array is returned and
       #         a warning is printed.
       #
       # @note   For compatibility with non Ruby clients a strings are used
       #         instead of symbols for the keys.
       #
-      # @return [Hash{String=>String}] A hash with the search information.
+      # @return [Array<String>] An array of words contained by the set's search related information.
       #
-      def search_data_from_set(set)
-        result = {}
+      def word_list_from_set(set)
         spec = set.specification
-        result['version'] = spec.version.to_s
-        result['summary'] = spec.summary
-        result['description'] = spec.description
-        result['authors'] = spec.authors.keys.sort * ', '
-        result
+        word_list = [set.name.dup]
+        if spec.summary
+          word_list += spec.summary.split
```

Comment:
Note that the default split only splits on whitespaces. We may need to be cleverer and for example also split on other characters. One idea is to `scan` instead of split. Try this example: `ruby -e 'p "Hello. This-is/an!Example".scan(/\w+/)'` (I am sorry for having suggested `split` earlier, only to now suggest `scan` – `split` also works, of course)

I believe we need a spec for this method, since the index data is so crucial for later searching.